### PR TITLE
Update GPT-2 Model List

### LINKED ISSUE #169: Support GPT-2 774M checkpoint file & Rename the current GPT2 checkpoint files
https://github.com/openai/gpt-2

In their official website, `Note that our original parameter counts were wrong due to an error (in our previous blog posts and paper). Thus you may have seen small referred to as 117M and medium referred to as 345M.`

We should rename ours as well. `We have currently released small (124M parameter), medium (355M parameter), and large (774M parameter) versions of GPT-2.`

### PR DESCRIPTION
**Pull request by gpengzhi**
Resolve #169

### CODE REVIEW

File: texar/torch/modules/pretrained/pretrained_gpt2.py [hunk 1]
```diff
@@ -46,18 +46,36 @@ class PretrainedGPT2Mixin(PretrainedMixin, ABC):
...
+                        for file in _CHECKPOINT_FILES],
+        'gpt2-large': [_GPT2_PATH + f"774M/{file}"
+                       for file in _CHECKPOINT_FILES],
     }
 
+    # Raise warning for the deprecated pre-trained model names
+    class MyDict(dict):
+        def __contains__(self, key):
+            if key == '117M':
+                raise KeyError("Pre-trained model name '117M' is deprecated, "
```
Comment (huzecong) [inline texar/torch/modules/pretrained/pretrained_gpt2.py]: Raising a `KeyError` is not really a "deprecation warning". Let's user the `warnings` module for that.

**Comment by gpengzhi** [comment]
> `raise`ing anything results in exceptions being thrown. You need `warnings.warn` and specify the type `UserWarning` there.

I see. Maybe we can use `DeprecationWarning` instead?

**Comment by huzecong** [comment]
`DeprecationWarning` is ignored by default, and they're intended towards developers. Basically, any warning that we intend the end-user to see should be `UserWarning`. Please refer to https://docs.python.org/3/library/warnings.html#warning-categories

**Comment by gpengzhi** [comment]
> `DeprecationWarning` is ignored by default, and they're intended towards developers. Basically, any warning that we intend the end-user to see should be `UserWarning`. Please refer to https://docs.python.org/3/library/warnings.html#warning-categories

Thanks for the explanation!

**Comment by huzecong** [comment]
Now the user will get a warning when using `115M` or `337M`, but they can't really use it right? `__contains__` returns `None` for these two keys, and there's actually no values corresponding to them.

**Comment by gpengzhi** [comment]
I must misunderstand the definition of warning. I will modify it accordingly.

**Comment by gpengzhi** [comment]
My bad. I thought there is no big difference between raising exceptions and raising warnings.

**Pull request merged**